src/Microsoft.PowerShell.Commands.Utility/commands/utility/AddType.cs

```diff
@@ -1010,77 +850,356 @@ private Assembly LoadAssemblyHelper(string assemblyName)
+                foreach (string assembly in ReferencedAssemblies)
+                {
+                    if (string.IsNullOrWhiteSpace(assembly)) { continue; }
+                    string resolvedAssemblyPath = ResolveAssemblyName(assembly, true);
+
+                    // Ignore some specified reference assemblies
+                    string fileName = PathType.GetFileName(resolvedAssemblyPath);
+                    if (s_refAssemblyNamesToIgnore.Value.Contains(fileName))
+                    {
+                        WriteVerbose(StringUtil.Format(AddTypeStrings.ReferenceAssemblyIgnored, resolvedAssemblyPath));
+                        continue;
+                    }
+                    tempReferences.Add(MetadataReference.CreateFromFile(resolvedAssemblyPath));
+                }
+                return tempReferences;
+            }
+            else
+            {
+                return s_defaultAssemblies.Value;
+            }
         }
 
         private void WriteTypes(Assembly assembly)
         {
             WriteObject(assembly.GetTypes(), true);
         }
 
-        private void CompileSourceToAssembly(string source)
+        #endregion LoadAssembly
+
+        #region SourceCodeProcessing
+
+        private OutputKind OutputAssemblyTypeToOutputKind(OutputAssemblyType outputType)
+        {
+            switch (outputType)
+            {
+                case OutputAssemblyType.Library:
+                    return OutputKind.DynamicallyLinkedLibrary;
+                case OutputAssemblyType.ConsoleApplication:
+                    return OutputKind.ConsoleApplication;
+                case OutputAssemblyType.WindowsApplication:
+                    return OutputKind.WindowsApplication;
+                default:
+                    throw new ArgumentOutOfRangeException("outputType");
+            }
+        }
+
+        private CommandLineArguments DefaultArgsParse(IEnumerable<string> args)
         {
-            CSharpParseOptions parseOptions;
-            if (Language == Language.CSharp)
+            string sdkDirectory = s_defaultSdkDirectory;
+            string baseDirectory = this.SessionState.Path.CurrentLocation.Path;
+            string additionalReferenceDirectories = null;
+
+            switch (Language)
+            {
+                case Language.CSharp:
+                    return CSharpCommandLineParser.Default.Parse(args, baseDirectory, sdkDirectory, additionalReferenceDirectories);
+                case Language.VisualBasic:
+                    return VisualBasicCommandLineParser.Default.Parse(args, baseDirectory, sdkDirectory, additionalReferenceDirectories);
+                default:
+                    Diagnostics.Assert(false, "ParseText: Unsupported language family.");
+                    break;
+            }
+
+            return null;
+        }
+
+        private SyntaxTree ParseText(SourceText sourceText, ParseOptions parseOptions, string path = "")
```

Comment:
nit: Maybe rename this method to `ParseSourceText`? It's more readable.

Reply:
Renamed.